args.go

```diff
@@ -623,3 +623,12 @@ func decodeArgAppendNoPlus(dst, src []byte) []byte {
 	}
 	return dst
 }
+
+func peekAllArgBytesToDst(dst *[][]byte, h []argsKV, k []byte) {
+	for i, n := 0, len(h); i < n; i++ {
+		kv := &h[i]
+		if bytes.Equal(kv.key, k) {
+			*dst = append(*dst, kv.value)
+		}
+	}
+}
```

Comment:
Sorry one minor last change. I don't think it's needed to pass a pointer to the `[][]byte`. Just append to it locally and return it. That's the pattern that is used in a lot of other places as well. It shouldn't cause any allocations.

```suggestion
func peekAllArgBytesToDst(dst [][]byte, h []argsKV, k []byte) [][]byte {
	for i, n := 0, len(h); i < n; i++ {
		kv := &h[i]
		if bytes.Equal(kv.key, k) {
			dst = append(dst, kv.value)
		}
	}
	return dst
}
```

Reply:
Thanks and done